packages/main/src/plugin/index.ts

```diff
@@ -2153,6 +2187,26 @@ export class PluginSystem {
       },
     );
 
+    this.ipcHandle('task:cancel', async (_listener, taskId: string): Promise<void> => {
+      const task = taskManager.getTask(taskId);
+      // Ensure the task is cancellable
+      if (!task?.cancellableTokenId) return;
+
+      const tokenSource = cancellationTokenRegistry.getCancellationTokenSource(task.cancellableTokenId);
+      // If the token source exist
+      if (tokenSource) {
+        // We cancel it
+        tokenSource.cancel();
+        // Update the task accordinaly
+        if (taskManager.isStatefulTask(task)) {
+          task.state = 'completed';
+          task.status = 'cancelled';
+          task.error = 'task cancelled';
+        }
+        taskManager.updateTask(task);
```

Comment:
you could call the cancelTask function from here to avoid code duplication. And maybe renaming it as it does not really cancel a task but it just update the task state

Reply:
Agreed !